```
08:53:24  [ERROR] /home/jenkins/agent/workspace/eclipse.jdt.ui-github_PR-2930/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/compare/JavaMergeViewer.java:[88] 
08:53:24  [ERROR] 	import org.eclipse.jdt.internal.ui.javaeditor.SemanticHighlightings;
08:53:24  [ERROR] 	       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
08:53:24  [ERROR] The import org.eclipse.jdt.internal.ui.javaeditor.SemanticHighlightings is never used
```

--- a/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/compare/JavaMergeViewer.java
+++ b/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/compare/JavaMergeViewer.java
@@ -622,6 +622,10 @@ public class JavaMergeViewer extends TextMergeViewer {
 			super.doSetInput(input);
 			// the editor input has been explicitly set
 			fInputSet = true;
+
+			if (getInputJavaElement() != null) {
+				installSemanticHighlighting();
+			}
 		}
 		// called by org.eclipse.ui.texteditor.TextEditorAction.canModifyEditor()
 		@Override

